async-nats/src/options.rs

```diff
@@ -429,154 +423,21 @@ impl ConnectOptions {
     /// # #[tokio::main]
     /// # async fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
     /// let (tx, mut _rx) = tokio::sync::mpsc::channel(1);
-    /// async_nats::ConnectOptions::new().error_callback(move |error| {
+    /// async_nats::ConnectOptions::new().event_callback(move |event| {
     ///     let tx = tx.clone();
     ///     async move {
-    ///         tx.send(error).await.unwrap();
+    ///         tx.send(event).await.unwrap();
     ///         }
     /// }).connect("demo.nats.io").await?;
     /// # Ok(())
     /// # }
     /// ```
-    pub fn error_callback<F, Fut>(mut self, cb: F) -> ConnectOptions
+    pub fn event_callback<F, Fut>(mut self, cb: F) -> ConnectOptions
     where
-        F: Fn(ServerError) -> Fut + Send + Sync + 'static,
+        F: Fn(Event) -> Fut + Send + Sync + 'static,
         Fut: Future<Output = ()> + 'static + Send + Sync,
     {
-        self.error_callback =
-            CallbackArg1::<ServerError, ()>(Box::new(move |error| Box::pin(cb(error))));
```

Comment:
Change the type to something like a concrete EventFn type?

Reply:
Was about removing it, but realized that this type is also used for JWT. Let's keep it.